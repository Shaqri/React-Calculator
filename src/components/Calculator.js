import React, { useState } from 'react';
import Navbar from './Navbar';
import Calculate from '../logic/calculate';
import * as calcStyles from '../styles/calculator.module.css';
import * as styles from '../styles/components.module.css';
import Header from './Header';
import Display from './Display';
import ButtonPanel from './ButtonPanel';

const Calculator = () => {
  const [total, setTotal] = useState(null);
  const [operation, setOperation] = useState(null);
  const [next, setNext] = useState(null);

  const handleClick = buttonName => {
    const newState = Calculate({ total, next, operation }, buttonName);
    setTotal(newState.total);
    setOperation(newState.operation);
    setNext(newState.next);
  };

  return (
    <>
      <Navbar />
      <div className={`${styles.container} ${styles.flexRow}`} data-testid="calculator-container">
        <Header text="Let's do some math!" />
        <div className={calcStyles.calculator}>
          <Display result={next || total || '0'} />
          <ButtonPanel handleClick={handleClick} />
        </div>
      </div>
    </>

  );
};

export default Calculator;
