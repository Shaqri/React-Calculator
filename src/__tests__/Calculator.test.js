import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import Calculator from '../components/Calculator';
/* eslint-disable react/display-name */
jest.mock('../components/Navbar', () => () => <nav data-testid="navbar" />);
jest.mock('../components/Header', () => () => <nav data-testid="header" />);
jest.mock('../components/Display', () => () => <nav data-testid="display" />);
jest.mock('../components/ButtonPanel', () => () => <nav data-testid="buttonPanel" />);

describe('Calculator', () => {
  it('renders navbar component', async () => {
    render(<Calculator />);
    const navbar = await screen.getByTestId(/navbar/);
    expect(navbar).toBeInTheDocument();
    expect(navbar).toMatchSnapshot();
  });

  it('renders calculator container', async () => {
    render(<Calculator />);
    const container = await screen.getByTestId(/calculator-container/);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders Header component within calculator container', async () => {
    render(<Calculator />);
    const header = await screen.getByTestId(/header/);
    expect(header).toBeInTheDocument();
    expect(header).toMatchSnapshot();
  });

  it('renders Display component within calculator container', async () => {
    render(<Calculator />);
    const display = await screen.getByTestId(/display/);
    expect(display).toBeInTheDocument();
    expect(display).toMatchSnapshot();
  });

  it('renders ButtonPanel component within calculator container', async () => {
    render(<Calculator />);
    const buttonPanel = await screen.getByTestId(/buttonPanel/);
    expect(buttonPanel).toBeInTheDocument();
    expect(buttonPanel).toMatchSnapshot();
  });
});
