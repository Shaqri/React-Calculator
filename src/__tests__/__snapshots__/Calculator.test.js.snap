// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders ButtonPanel component within calculator container 1`] = `
<nav
  data-testid="buttonPanel"
/>
`;

exports[`Calculator renders Display component within calculator container 1`] = `
<nav
  data-testid="display"
/>
`;

exports[`Calculator renders Header component within calculator container 1`] = `
<nav
  data-testid="header"
/>
`;

exports[`Calculator renders calculator container 1`] = `
<div
  class="undefined undefined"
  data-testid="calculator-container"
>
  <nav
    data-testid="header"
  />
  <div>
    <nav
      data-testid="display"
    />
    <nav
      data-testid="buttonPanel"
    />
  </div>
</div>
`;

exports[`Calculator renders navbar component 1`] = `
<nav
  data-testid="navbar"
/>
`;
